### DISCUSSION TITLE
New Device isn't so new any more.   Can you automatically class as no longer "new" after x period?

### DISCUSSION THREAD

**Discussion by thefunkygibbon** [netalertx/NetAlertX#795] [General]
I thought this was an option, but I've waded through the new device settings and it cannot see anything particularly obvious relating to this.   
I would want devices which were first seen over a week ago to no longer be labelled as a "new device".    Am i going blind/mad here?

**Comment by jokob-sk** [comment]
Hey @thefunkygibbon ,

Saw your comment about disappearing devices -


> ok for some reason i've lost all of my  "offline" devices which i previously had in there. not sure if its anything to do with this dev update?
> 
> or could it be to do with changing the option of "Keep new devices for" (changed from 0 to 168). I figured it was the option which automatically de-tags it as being "new" (couldn't see any other option which appeared to do that). Am i going mad? (sorry to keep it in this post, I did put about it in the "discussions" page but didn't get a response.

@thefunkygibbon this option removes the device completely, not only the New device flag:

`HRS_TO_KEEP_NEWDEV` - This is a maintenance setting. If enabled (0 is disabled), **devices marked as New Device will be deleted** if their First Session time was older than the specified hours in this setting. Use this setting if you want to auto-delete New Devices after X hours. 

I will look into having the option to unmark the device as new

**Reply by jokob-sk** [reply]
@thefunkygibbon I just pushed an update to -dev that's adding a `CLEAR_NEW_FLAG` setting allowing you to auto-clear the NEW flag after X hours. It would be great if you could test it and confirm is working. The -dev image should be available to pull in about 15 min from now

**Comment by thefunkygibbon** [comment]
just checking it now.  couple of questions
the option that i previously used (delete new after....) seemed to have removed the items based on the time already spent in the system.  (ie if i put 168 hours, it deleted all of the devices which were offline and were first seen more than 168 hours ago).   
whereas this new option for "unflag new" hasn't removed any of the new tags yet (i have put it as 1 hour just to test) and all of the devices on my system were originally picked up weeks ago.   so i'll see in an hours time  if its based on the time i set the setting or if its not working. 

also, the devices which it deleted when i mistakenly used the "delete new after" option (what is even the usecase for this setting... surely the thing i asked for is much more relevant?!,  is there any way of recovering/undeleting them?  I assume they were old, previously connected devices which my omada would have informed it of when i first set it up.

**Reply by jokob-sk** [reply]
Thanks for testing! The maintenance task runs once every 30 minutes so it will take a bit time for the devices to be unselected as new.

The HRS_TO_KEEP_NEWDEV was a user request for what I think is a public WiFi to delete devices after a while.

You might be able to recover the deleted devices from one of the devices.csv (if available) files found in the /config folder. (use the tools under the maintenance section to recover the devices)

**Comment by thefunkygibbon** [comment]
this is great, it seems to work as expected, thanks.  out of interest, any reason why its in the "core" section of settings , rather than the "new devices" section which is where one would expect it to be :-)

**Reply by jokob-sk** [reply]
Thanks for checking! The NEWDEV plugin is only a template for new devices as per description. Any backend-related processes are under core or respective plugins. Hope that makes sense!

**Reply by sdutremble** [reply]
Just an observarion: instead of deleting anything, archive may be more appropriate.
